tests/test_sentence/test_corpus/test_tei_input.py

```diff
@@ -52,22 +54,69 @@ def test_attributes(self):
         assert ms_page.number == "IX"
         assert ms_page.edition == "manuscript"
 
-    def test_str(self):
+    @patch.object(TEIPage, "get_body_text")
+    @patch.object(TEIPage, "get_footnote_text")
+    def test_str(self, mock_get_footnote_text, mock_get_body_text):
+        mock_get_body_text.return_value = "Mock body text"
+        mock_get_footnote_text.return_value = "Mock footnote text"
+
+        tei_doc = TEIDocument.init_from_file(TEST_TEI_FILE)
+        page = tei_doc.all_pages[0]
+
+        result = str(page)
+
+        # Verify mocks were called (may be called multiple times)
+        assert mock_get_body_text.called
+        assert mock_get_footnote_text.called
+
+        # should return both body text and footnote text, separated by double newlines
+        assert result == "Mock body text\n\nMock footnote text"
+
+    def test_get_body_text_no_footnotes(self):
         tei_doc = TEIDocument.init_from_file(TEST_TEI_FILE)
         # test first page
         page = tei_doc.all_pages[0]
         # includes some leading whitespace from <pb> and <p> tags
         # remove whitespace for testing for now
-        text = str(page).strip()
+        text = page.get_body_text()
 
+        # Should not contain leading or trailing whitespace
+        assert text == text.strip()
         # first text content after the pb tag
         assert text.startswith("als in der ersten Darstellung.")  # codespell:ignore
         # last text content after the next standard pb tag
         assert text.endswith("entwickelten nur das Bild der eignen Zukunft!")
         # should not include editorial content
         assert "|" not in text
         assert "IX" not in text
-        # TODO: eventually should not include footnote content
+
+    def test_get_body_text_with_footnotes(self):
```

Comment:
I was confused by the method name. Suggest adding a comment to clarify:
```suggestion
    def test_get_body_text_with_footnotes(self):
         # test a sample page with footnotes to confirm footnote contents are excluded
```